controllers/v1/observationsController.js

```diff
@@ -2160,4 +2247,53 @@ module.exports = class Observations extends Abstract {
         })
     }
 
-}
\ No newline at end of file
+
+}
+
+/**
+  * get subEntities of matching type by recursion.
+  * @method
+  * @name getSubEntitiesBasedOnEntityType
+  * @param {Array} parentEntities - Array of entity Ids - parent entities.
+  * @param {String} entityType - Entity type.
+  * @param {Array} matchingData - subentities of type {entityType} of {parentEntities}
+  * @returns {Array} - Sub entities matching the type.
+*/
+
+async function getSubEntitiesBasedOnEntityType( parentEntities, entityType, matchingData ) {
+    //get sub entities of type {entityType} of entities {parentEntity}
+    if( !parentEntities.length > 0 ){
+      return matchingData;
+    }
+
+    let bodyData={
+        "parentId" : parentEntities
+    };
+    
+    let entityDetails = await userProfileService.locationSearch(bodyData);
+
+    if( ( !entityDetails.success ) && !matchingData.length > 0 ) {
+      return matchingData;
+    }
+    
+    let entityData = entityDetails.data;  
+    let mismatchEntities = [];
+
+    entityData.map(entity => {
+      if( entity.type == entityType ) {
+        matchingData.push(entity)
+      } else {
+        mismatchEntities.push(entity.id)
+      }
+    });
+
+    if( mismatchEntities.length > 0 ){
+      await getSubEntitiesBasedOnEntityType(mismatchEntities, entityType, matchingData)
+    } 
+    let uniqueEntities = [];
```

Comment:
@VISHNUDAS-tunerlabs as discussed this is not required. Also please make variable name changes.